src/v/cloud_storage/topic_manifest.h

```diff
@@ -22,6 +22,14 @@ namespace cloud_storage {
 struct topic_manifest_handler;
 class topic_manifest final : public base_manifest {
 public:
+    constexpr static int32_t first_version = 1;
+    // this version introduces the serialization of
+    // cluster::topic_configuration, with all its field
+    constexpr static int32_t cluster_topic_configuration_version = 2;
+
+    constexpr static int32_t current_version
```

Comment:
Proposing a codedoc: `Version used to identify the current serialization format and the maximum version which can be deserialized from.`